### DISCUSSION TITLE
GH Actions fails to pass ENV VARS between steps!?

### DISCUSSION THREAD

**Discussion by viliusddd** [community/community#25290] [Actions]
<p>Hi,<br>
I can’t get it to pass variables between jobs. Any ideas??</p>
<pre><code class="lang-yaml">name: Example Workflow
on: [push]

jobs:
  example-job:
    name: An example job
    runs-on: ubuntu-latest
    steps:
      - run: echo $HOME
        env:
          FOO: 'BAZ'

      - run: |
          echo $FOO
          echo ${{ env.FOO }}
</code></pre>
<p>output:<br>
<img src="https://user-images.githubusercontent.com/30282958/181090099-7b4c64db-d8a5-4191-b670-bef15384756e.png" alt="Screenshot_13" data-base62-sha1="6hggy8stKKENPbFuep6iTy1DRn0" width="275" height="469"></p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="viliusddd" data-post="1" data-topic="230828">
<div class="title">
<div class="quote-controls"></div>
 viliusddd:</div>
<blockquote>
<pre><code class="lang-yaml">      - run: echo $HOME
        env:
          FOO: 'BAZ'
</code></pre>
</blockquote>
</aside>
<p>This is working as intended: If you set an environment variable for a single step, it’s only available in that step. This is useful for example to limit the visibility of secrets.</p>
<p>If you want to set the environment variable for a whole job, you can either statically <a href="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idenv">set it in the job</a>, or dynamically by <a href="https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable">writing to <code>$GITHUB_ENV</code></a>. With the second option the environment variable will be available to all steps after the current one within the job.</p>
